Sources/Danger/GitHubDSL.swift

```diff
@@ -0,0 +1,329 @@
+    public let label: String
+
+    /// The reference point for the merge, e.g. "master"
+    public let ref: String
+
+    /// The reference point for the merge, e.g. "704dc55988c6996f69b6873c2424be7d1de67bbe"
+    public let sha: String
+
+    /// The user that owns the merge reference e.g. "artsy"
+    public let user: GitHubUser
+
+    /// The repo from which the reference comes from
+    public let repo: GitHubRepo
+
+}
+
+// MARK: - GitHubRepo
+
+public struct GitHubRepo: Decodable {
+
+    // MARK: - CodingKeys
+
+    enum CodingKeys: String, CodingKey {
+        case id
+        case name
+        case owner
+        case description
+        case fullName = "full_name"
+        case isPrivate = "private"
+        case isFork = "fork"
+        case htmlURL = "html_url"
+    }
+
+    // MARK: - Properties
+
+    /// Generic UUID.
+    public let id: Int
+
+    /// The name of the repo, e.g. "danger-swift".
+    public let name: String
+
+    /// The full name of the owner + repo, e.g. "Danger/danger-swift"
+    public let fullName: String
+
+    /// The owner of the repo.
+    public let owner: GitHubUser
+
+    /// A boolean stating whether the repo is publicly accessible.
+    public let isPrivate: Bool
+
+    /// A textual description of the repo.
+    public let description: String
+
+    /// A boolean stating whether the repo is a fork.
+    public let isFork: Bool
+
+    /// The root web URL for the repo, e.g. https://github.com/artsy/emission
+    public let htmlURL: String
+
+}
+
+// MARK: - GitHubReview
+
+public struct GitHubReview: Decodable {
+
+    // MARK: - ReviewState
+
+    public enum ReviewState: String, Decodable {
+        case approved
+        case requestedChanges
+        case comment
+        case pending
+    }
+
+    /// The user who has completed the review or has been requested to review.
+    public let user: GitHubUser
+
+    /// The id for the review (if a review was made).
+    public let id: Int?
+
+    /// The body of the review (if a review was made).
+    public let body: String?
+
+    /// The commit ID the review was made on (if a review was made).
+    public let commitID: String?
+
+    /// The state of the review (if a review was made).
+    public let state: ReviewState?
+
+}
+
+// MARK: - GitHubCommit
+
+/// A GitHub specific implementation of a git commit.
+public struct GitHubCommit: Decodable {
+
+    // MARK: - Properties
+
+    /// The SHA for the commit.
+    public let sha: String
+
+    /// The raw commit metadata.
+    public let commit: GitCommit
+
+    /// The URL for the commit on GitHub.
+    public let url: String
+
+    /// The GitHub user who wrote the code.
+    public let author: GitHubUser
+
+    /// The GitHub user who shipped the code.
+    public let committer: GitHubUser
+
+}
+
+// MARK: - GitHubIssue
+
+public struct GitHubIssue: Decodable {
+
+    // MARK: - Properties
+
+    /// The labels associated with this issue.
+    public let labels: [GitHubIssueLabel]
+
+}
```

Comment:
Because of JS' nature of interfaces I can get away with just including the labels on this struct, but realistically it might be worth typing the entire thing - I doubt its going to change in the future

Reply:
Here's what it looks like: https://github.com/danger/danger-js/blob/master/source/platforms/_tests/fixtures/github_issue.json

Reply:
I was curious about that lol. I thought it could be because they could access similar data elsewhere. Not a problem -- I will make these changes 👍 Swift 4 JSON decoding is super easy.